test/dotnet-add-p2p.Tests/GivenDotnetAddP2P.cs

```diff
@@ -626,5 +642,79 @@ public void WhenReferenceIsRelativeAndProjectIsNotInCurrentDirectoryAndForceSwit
             csproj.NumberOfItemGroupsWithoutCondition().Should().Be(noCondBefore + 1);
             csproj.NumberOfProjectReferencesWithIncludeContaining(setup.ValidRefCsprojRelPath.Replace('/', '\\')).Should().Be(1);
         }
+
+        [Fact]
+        public void ItCanAddRefWithCondOnCompatibleFramework()
+        {
+            var setup = Setup();
+            var lib = new ProjDir(setup.LibDir);
+            var net45lib = new ProjDir(Path.Combine(setup.TestRoot, "Net45Lib"));
+
+            int condBefore = lib.CsProj().NumberOfItemGroupsWithConditionContaining(ConditionFrameworkNet451);
```

Comment:
So this is actually about ItemGroups - this counts also item groups with other stuff. I'd have to make that method more clever to call it the way you propose - that it is a good idea but I believe there is slightly more work involved as it is already used in 3 test projects. we can do it as a post improvement